### PR TITLE
Custom `dist` overrides version build number

### LINKED ISSUE #210: `SENTRY_DIST` env. variable doesn't work
### Environment

1.7.1

### Steps to Reproduce

I have the following script:

```bash
SENTRY_DIST=$BUILD_NUMBER
echo "Sentry distribution number: $SENTRY_DIST"
dart run sentry_dart_plugin
```

this prints:

```console
Sentry distribution number: 14
```

But after the `sentry_dart_plugin` runs I see this:

```console
Created release my_app@0.0.1+1
```

So it doesn't seem to me that `SENTRY_DIST` is being used by the CLI, as I am expecting to see `my_app@0.0.1+14`.

### PR DESCRIPTION
**Pull request by denrase**
## :scroll: Description

When a custom `dist` is provided, either though pubspec or env variable, it takes precedence over the versions build number.

## :bulb: Motivation and Context

Closes #210

## :green_heart: How did you test it?

Unit tests

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: lib/src/configuration.dart [hunk 1]
```diff
@@ -51,11 +51,11 @@ class Configuration {
   /// The Apps release name, defaults to 'name@version+buildNumber' from pubspec or set via env. var. SENTRY_RELEASE
   /// Example, name: 'my_app', version: 2.0.0+1, in this case the release is my_app@2.0.0+1
   /// This field has precedence over the [name] from pubspec
-  /// If this field has a build number, it has precedence over the [version]'s buid number from pubspec
+  /// If this field has a build number, it has precedence over the [version]'s build number from pubspec
   late String? release;
 
-  /// The Apps dist/build number, defaults to the build number after the '+' char from [version]'s pubspec or ser via env. var. SENTRY_DIST
-  /// Example, version: 2.0.0+1, in this case the build number is 1
+  /// The Apps dist/build number, taken from pubspec dist or SENTRY_DIST env. variable
```
Comment (buenaflor) [inline lib/src/configuration.dart:57]: I assume sentry-defines should also be included here as soon as it's merged right?

File: lib/src/configuration.dart [hunk 2]
```diff
@@ -51,11 +51,11 @@ class Configuration {
   /// The Apps release name, defaults to 'name@version+buildNumber' from pubspec or set via env. var. SENTRY_RELEASE
```
Comment (buenaflor) [inline lib/src/configuration.dart:51]: same here about sentry-defines

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## Unreleased
+
+### Features
+
+- Custom `dist` overrides version build number ([#216](https://github.com/getsentry/sentry-dart-plugin/pull/216))
```
Comment (buenaflor) [inline CHANGELOG.md:7]:
```suggestion
- Custom `dist` overrides version build number ([#216](https://github.com/getsentry/sentry-dart-plugin/pull/216))
  - For instance, if the initial release version is `release@1.0.0+1`, specifying a custom dist value of 2 will update the version to `release@1.0.0+2`. 
```

Added some more info for more clarity, you can modify it if you think there's a better wording

**Review by buenaflor — APPROVED** [review]
Looks good, only got some minor comments

File: lib/src/configuration.dart (hunk 1 above)
Comment (denrase) [reply lib/src/configuration.dart:57]: Yes, probably best to do this from the sentry-defines PR

**Comment by buenaflor** [comment]
@denrase looks good here if you wanna merge

**Pull request merged**